composeApp/src/commonMain/kotlin/com/github/familyvault/ui/screens/main/familyGroupSettings/familyGroupMember/ModifyFamilyMemberScreen.kt

```diff
@@ -78,8 +105,15 @@ class ModifyFamilyMemberScreen(private val familyMember: FamilyMember) : Screen
         )
         var savingChanges by remember { mutableStateOf(false) }
         var showDialog by remember { mutableStateOf(false) }
-        var selectedPermissionGroup by remember { mutableStateOf(FamilyGroupMemberPermissionGroup.Guest) }
-        val coroutineScope = rememberCoroutineScope()
+        var selectedPermissionGroup by remember { mutableStateOf(familyMember.permissionGroup) }
+
+        val isGuardian = currentUserPermissionGroup == FamilyGroupMemberPermissionGroup.Guardian
+
+        fun isOptionDisabled(option: PermissionOption): Boolean {
+            return isLastGuardian && 
+                   familyMember.permissionGroup == FamilyGroupMemberPermissionGroup.Guardian &&
+                   option.permissionGroup != FamilyGroupMemberPermissionGroup.Guardian
+        }
```

Comment:
Co robi ta funkcja? Ja bym może zmienił `isOptionDisabled` na to co jest tutaj sprawdzane aby było bardziej czytelnie